spring-integration-core/src/main/java/org/springframework/integration/endpoint/PollingConsumer.java

```diff
@@ -71,15 +91,88 @@ protected void doStop() {
 
 	@Override
 	protected void handleMessage(Message<?> message) {
-		this.handler.handleMessage(message);
+		boolean executorInterceptorPresent = false;
+		if (this.channelInterceptors != null) {
+			for (ChannelInterceptor interceptor : this.channelInterceptors) {
+				if (interceptor instanceof ExecutorChannelInterceptor) {
+					executorInterceptorPresent = true;
```

Comment:
I think we need an optimization to avoid iterating over the interceptors to see if we have an exec interceptor.

Maybe a sub interface of `CIA` - `ExecutorChannelInterceptorAware.hasExecInterceptor()` ?

We can still get the list up front, but instead of iterating, just check the boolean.